include/freerdp/freerdp.h

```diff
@@ -360,6 +361,9 @@ FREERDP_API void freerdp_free(freerdp* instance);
 FREERDP_API BOOL freerdp_focus_required(freerdp* instance);
 FREERDP_API void freerdp_set_focus(freerdp* instance);
 
+FREERDP_API int freerdp_get_dpu_reason(rdpContext* context);
+FREERDP_API void freerdp_set_dpu_reason(rdpContext* context, int reason);
```

Comment:
are you sure the setter needs to be exported? This is a protocol internal variable, so this should be hidden.

Reply:
Good point.  I was just making it symmetric without thinking.  I can just remove the setter altogether.  There don't seem to be many others anyway, so that would be consistent.

Reply:
@r-barnett maybe `freerdp_get_disconnect_ultimatum`? seems short enough and descriptive